lib/screens/episode.dart

```diff
@@ -269,55 +279,66 @@ class _EpisodeScreenState extends ConsumerState<EpisodeScreen> with AutoRouteAwa
       ),
     ];
 
-    showModalBottomSheet(
-      useRootNavigator: true,
-      context: context,
-      builder: (ctx) => BottomSheetSelect(
-        title: S.of(context).share,
-        description: episode.shareRestriction == Enum$ShareRestriction.public
-            ? null
-            : Padding(
-                padding: EdgeInsets.only(bottom: 16),
-                child: Row(
-                  children: [
-                    Padding(padding: EdgeInsets.all(4), child: SvgPicture.string(SvgIcons.infoCircle)),
-                    Expanded(
-                      child: Padding(
-                        padding: EdgeInsets.only(left: 8, right: 16),
-                        child: Text(
-                          'This video is only accessible to users that are logged in to the app.',
-                          style: BtvTextStyles.caption1.copyWith(color: BtvColors.label2),
-                        ),
+    var builder = BottomSheetSelect(
+      title: S.of(context).share,
+      description: episode.shareRestriction == Enum$ShareRestriction.public
+          ? null
+          : Padding(
+              padding: const EdgeInsets.only(bottom: 16),
+              child: Row(
+                children: [
+                  Padding(padding: const EdgeInsets.all(4), child: SvgPicture.string(SvgIcons.infoCircle)),
+                  Expanded(
+                    child: Padding(
+                      padding: const EdgeInsets.only(left: 8, right: 16),
+                      child: Text(
+                        'This video is only accessible to users that are logged in to the app.',
+                        style: BtvTextStyles.caption1.copyWith(color: BtvColors.label2),
                       ),
-                    )
-                  ],
-                ),
+                    ),
+                  )
+                ],
               ),
-        selectedId: 'fromStart',
-        items: options,
-        showSelection: false,
-        onSelectionChanged: (id) {
-          var episodeUrl = 'https://app.bcc.media/episode/${widget.episodeId}';
+            ),
+      selectedId: 'fromStart',
+      items: options,
+      showSelection: false,
+      onSelectionChanged: (id) {
+        var episodeUrl = 'https://app.bcc.media/episode/${widget.episodeId}';
+
+        Future shareFuture = Future(() async {
+          showModalBottomSheet(
```

Comment:
Doesn't this animate in a new bottom sheet? If not, can you upload a screen recording of how this looks?

Reply:
I would rather make a stateful component called `ShareEpisodeBottomSheet` in a different file (components/share_episode_bottom_sheet.dart) and move all the logic in there.